persist-cli/src/main/java/io/ballerina/persist/cmd/Migrate.java

```diff
@@ -559,16 +598,70 @@ private static List<String> findDifferences(Module previousModel, Module current
+                convertMapToMySQLQuery(QueryTypes.ADD_READONLY, addedReadOnly, queries, addedEntities);
+                convertFKMapToMySQLQuery(QueryTypes.ADD_FOREIGN_KEY, addedForeignKeys, queries);
+                convertMapListToMySQLQuery(QueryTypes.REMOVE_FIELD, removedFields, queries);
+                convertListToMySQLQuery(QueryTypes.REMOVE_TABLE, removedEntities, queries);
+                convertMapToMySQLQuery(QueryTypes.CHANGE_TYPE, changedFieldTypes, queries, addedEntities);
+                break;
+
+            case("googlesheets"):
+                convertCreateTableToGSheetsQuery(QueryTypes.ADD_TABLE, 
+                    addedEntities, queries, addedReadOnly, addedFields);
+                List<String> addFieldWarnings = convertMapToGSheetsQuery(QueryTypes.ADD_FIELD, 
+                    addedFields, queries, addedEntities);
+                for (String warning : addFieldWarnings) {
+                    warnings.add(warning);
+                }
+                List<String> removeFKWarnings = convertMapListToGSheetsQuery(QueryTypes.REMOVE_FOREIGN_KEY, 
+                    removedForeignKeys, queries);
+                for (String warning : removeFKWarnings) {
+                    warnings.add(warning);
+                }
+                List<String> removePKWarnings = convertMapListToGSheetsQuery(QueryTypes.REMOVE_READONLY, 
+                    removedReadOnly, queries);
+                for (String warning : removePKWarnings) {
+                    warnings.add(warning);
+                }
+                List<String> addReadOnlyWarnings = convertMapToGSheetsQuery(QueryTypes.ADD_READONLY, 
+                    addedReadOnly, queries, addedEntities);
+                for (String warning : addReadOnlyWarnings) {
+                    warnings.add(warning);
+                }
+                List<String> addFKWarnings = convertFKMapToGSheetsQuery(QueryTypes.ADD_FOREIGN_KEY, 
+                    addedForeignKeys, queries);
+                for (String warning : addFKWarnings) {
+                    warnings.add(warning);
+                }
+                List<String> removeFieldWarnings = convertMapListToGSheetsQuery(QueryTypes.REMOVE_FIELD, 
+                    removedFields, queries);
+                for (String warning : removeFieldWarnings) {
+                    warnings.add(warning);
+                }
+                List<String> removeTableWarnings = convertListToGSheetsQuery(QueryTypes.REMOVE_TABLE, 
+                    removedEntities, queries);
+                for (String warning : removeTableWarnings) {
+                    warnings.add(warning);
+                }
+                List<String> changeDataTypeWarnings = convertMapToGSheetsQuery(QueryTypes.CHANGE_TYPE, 
+                    changedFieldTypes, queries, addedEntities);
+                for (String warning : changeDataTypeWarnings) {
+                    warnings.add(warning);
+                }
+                break;
```

Comment:
Better we use `Factory Pattern` instead of a switch case. As more data store comes in, this is not scalable and maintainable